app/grandchallenge/algorithms/views.py

```diff
@@ -208,6 +210,9 @@ def get_context_data(self, **kwargs):
             {
                 "pending_permission_requests": pending_permission_requests,
                 "github_app_install_url": f"{settings.GITHUB_APP_INSTALL_URL}?state={self.object.slug}",
+                "builds": Build.objects.filter(
```

Comment:
Is it not better to go the other way around and use `object.build_set.all()`? Then it can be used in a prefetch related too.

Reply:
Of course! That makes a lot more sense.

Reply:
Hmm upon further inspection that wouldn't work, as the build is linked to `AlgorithmImages` rather than the algorithm itself. So we'd have to go over the builds for each image linked to the algorithm. In which case I think the current query is more efficient.